packages/gitbook-docs/src/markdoc/schema/heading.ts

```diff
@@ -12,10 +13,25 @@ export const heading: Schema = {
     level: { type: Number, required: true },
   },
   transform(node, config) {
-    return new Markdoc.Tag(
-      `Heading`,
-      node.transformAttributes(config),
-      node.transformChildren({ ...config, nodes: { text: sanitizedText } })
-    );
+    const nodes = { text: sanitizedText };
+    const children = node.transformChildren({
+      ...config,
+      nodes,
+    });
+    const attributes = node.transformAttributes(config);
+
+    const id: string = children
+      .filter((child) => typeof child === 'string')
+      .map((child) =>
+        (child as string)
```

Comment:
Can we re-write it in a form that doesn't need [type-assertions](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions)?

Reply:
Do you mean something like this (I prefer the explicit cast to string)?
```suggestion
    const id: string = children
      .filter((child) => typeof child === 'string')
      .map((child) =>
        `${child}`
```

Would you happen to have any suggestions?

Reply:
What about?

``` typescript
children.map((child) => (typeof child === 'string' ? child : ''))
```

Reply:
I also added this filter `filter((child) => !!child)` to avoid extra `-` when joining empty strings